src/core/distributions/phylogenetics/tree/birthdeath/OccurrenceBirthDeathProcess.cpp

```diff
@@ -0,0 +1,543 @@
+#include <vector>
+
+#include "DistributionExponential.h"
+#include "OccurrenceBirthDeathProcess.h"
+#include "RandomNumberFactory.h"
+#include "RandomNumberGenerator.h"
+#include "RbConstants.h"
+#include "RbMathCombinatorialFunctions.h"
+#include "RbMathLogic.h"
+#include "AbstractBirthDeathProcess.h"
+#include "DagNode.h"
+#include "RbException.h"
+#include "RbVector.h"
+#include "TopologyNode.h"
+#include "Tree.h"
+#include "TypedDagNode.h"
+#include "StartingTreeSimulator.h"
+
+
+#include "ComputeLikelihoodsLtMt.h"
+
+namespace RevBayesCore { class Taxon; }
+
+using namespace RevBayesCore;
+
+/**
+ * Constructor.
+ * We delegate most parameters to the base class and initialize the members.
+ * \param[in]    sa                        Start age of the process.
+ * \param[in]    inspeciation              Speciation/birth rate(s).
+ * \param[in]    inextinction              Extinction/death rate(s).
+ * \param[in]    inserialsampling          Serial sampling rate(s).
+ * \param[in]    intreatment               Probabilit(y|ies) of death upon sampling (treatment).
+ * \param[in]    inoccurrence              Occurrence sampling rate(s).
+ * \param[in]    ineventsampling           Sampling probability at present time.
+ * \param[in]    ht                        Rate interval change times of the piecewise constant process.
+ * \param[in]    cdt                       Condition of the process (survival/survival2).
+ * \param[in]    tn                        Taxa.
+ * \param[in]    uo                        If true the start age is the origin time otherwise the root age of the process.
+ * \param[in]    t                         Starting tree if we want to avoid simulating trees.
+ * \param[in]    n                         Maximum number of hidden lineages (algorithm accuracy).
+ * \param[in]    O                         Vector of occurrence ages.
+
+ */
+OccurrenceBirthDeathProcess::OccurrenceBirthDeathProcess(                                  const TypedDagNode<double> *sa,
+                                                                                           const DagNode *inspeciation,
+                                                                                           const DagNode *inextinction,
+                                                                                           const DagNode *inserialsampling,
+                                                                                           const DagNode *intreatment,
+                                                                                           const DagNode *inoccurrence,
+                                                                                           const DagNode *ineventsampling,
+                                                                                           const TypedDagNode< RbVector<double> > *ht,
+                                                                                           const std::string &cdt,
```

Comment:
ideally, we would allow for either a joint timeline for all parameters, or one timeline per parameter. It also might be safer then to allow that the timeline is not sorted (see https://github.com/revbayes/revbayes/blob/development/src/core/distributions/phylogenetics/tree/birthdeath/EpisodicBirthDeathSamplingTreatmentProcess.cpp)